concordium-consensus/src/Concordium/GlobalState.hs

```diff
@@ -583,20 +589,43 @@ instance GlobalStateConfig DiskTreeDiskBlockConfig where
 
     initialiseExistingGlobalState _ DTDBConfig{..} = do
       -- check if all the necessary database files exist
-      existingDB <- checkExistingDatabase dtdbTreeStateDirectory dtdbBlockStateFile
-      pbscAccountCache <- liftIO $ Accounts.newAccountCache (rpAccountsCacheSize dtdbRuntimeParameters)
-      pbscModuleCache <- liftIO $ Modules.newModuleCache (rpModulesCacheSize dtdbRuntimeParameters)
-      if existingDB then do
-        pbscBlobStore <- liftIO $ do
-          -- the block state file exists, is readable and writable
-          -- we ignore the given block state parameter in such a case.
-          loadBlobStore dtdbBlockStateFile
-        let pbsc = PersistentBlockStateContext{..}
-        logm <- ask
-        skovData <- liftIO (runLoggerT (loadSkovPersistentData dtdbRuntimeParameters dtdbTreeStateDirectory pbsc) logm
-                    `onException` (closeBlobStore pbscBlobStore))
-        return (Just (pbsc, skovData))
-      else return Nothing
+      let treeStateFile = dtdbTreeStateDirectory </> "data.mdb"
```

Comment:
I am not sure why you had to inline the check. I feel the checking part was cleanly encapsulated in the function which could be slightly expanded to cover the deletion as well.

I urge you to reinstate the function since it also exists in the more appropriate place, in the treestate.

Reply:
As I explained [above](https://github.com/Concordium/concordium-node/pull/507#discussion_r977895389), I believe that a function should do one thing and do it well, in order to be reusable. If a function is used in one place only and there is little chance that it will be used elsewhere, it might as well be inlined to improve readability (one would not have to spread their attention between several modules).

But I have no problem with putting the check and the deletion in a separate function either. I'll do that.

Reply:
I agree with moving this functionality back to `checkExistingDatabase`. This module is intended to simply wrap up the functionality of the global state implementations, and most of the implementation should be elsewhere. The `GlobalStateConfig` implementations are supposed to be small, so that it would be easy to have other configurations with variants of the functionality. (e.g. historically there was a `DiskTreeDiskBlockWithLogConfig` that would share much of the implementation.)